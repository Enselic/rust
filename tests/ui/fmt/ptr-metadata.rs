//@ run-pass
//@ check-run-results
//@ normalize-stdout: "0x[0-9a-fA-F]+" -> "$$HEX"

use std::fmt::Display;

fn main() {
    let plain = &mut 777;
    println!("*mut i32: {:?}", plain as *mut i32);
    println!("*const i32: {:?}", plain as *const i32);

    let slice = &mut [1, 2, 3][..];
    println!("*mut [i32]: {:?}", slice as *mut [i32]);
    println!("*const [i32]: {:?}", slice as *const [i32]);

    let vtable = &mut 999 as &mut dyn Display;
    println!("*mut dyn Display: {:?}", vtable as *mut dyn Display);
    println!("*const dyn Display: {:?}", vtable as *const dyn Display);
}
