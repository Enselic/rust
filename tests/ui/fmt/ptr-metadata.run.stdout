*mut i32: $HEX
*const i32: $HEX
*mut [i32]: Pointer { addr: $HEX, metadata: 3 }
*const [i32]: Pointer { addr: $HEX, metadata: 3 }
*mut dyn Display: Pointer { addr: $HEX, metadata: DynMetadata($HEX) }
*const dyn Display: Pointer { addr: $HEX, metadata: DynMetadata($HEX) }
